cdap-app-fabric/src/main/java/co/cask/cdap/app/mapreduce/MRJobClient.java

```diff
@@ -71,13 +71,9 @@ public MRJobInfo getMRJobInfo(Id.Run runId) throws IOException, NotFoundExceptio
 
     JobClient jobClient;
     JobStatus[] jobs;
-    try {
-      jobClient = new JobClient(hConf);
-      jobs = jobClient.getAllJobs();
-    } catch (Exception e) {
-      LOG.warn("JobClient failed to get all jobs.", e);
-      throw new IOException(e);
-    }
+
+    jobClient = new JobClient(hConf);
+    jobs = jobClient.getAllJobs();
```

Comment:
getAllJobs can throw IOException when it gets a InterruptedException. We let it propagate up to its caller DistributedMRJobInfoFetcher where we have to catch it as in the case of failure we fail back to metric system. So, we catch it, log and return the info from metric system.